Lesson15/Akan/script.js

```diff
@@ -0,0 +1,407 @@
+/*
+4. Extract Domain
+   - Define a function `getDomain(email)` that uses `slice` or `substring` to
+     extract everything after '@'.
+   - Log the domain to the console.
+   - Example: "user@example.com" -> "example.com"
+*/
+
+function getDomain(email) {
+  let domain = email.slice(email.indexOf('@') + 1);
+  console.log(domain);
+}
+
+/*
+5. Check Substring
+   - Define a function `containsWord(sentence, word)` that checks if the `sentence`
+     includes `word` (use the .includes() method).
+   - If true, log: "<word> found in sentence."
+   - Else, log: "<word> not found in sentence."
+*/
+
+function containsWorld(sentence, world) {
+  if (sentence.includes(world)) {
+    console.log(world + " found in sentence.");
+  }
+  else {
+    console.log(world + " not found in sentence.");
+  }
+}
+
+/*
+6. File Extension Check
+   - Define a function `checkFileExtension(filename)` that checks if the filename
+     ends with ".pdf" using .endsWith().
+   - If it does, log: "This is a PDF file."
+   - Otherwise, log: "Not a PDF file."
+*/
+
+function checkFileExtension(filename) {
+  if (filename.endsWith('.pdf'))
+    console.log('This is a PDF file.');
+  else
+    console.log('Not a PDF file.');
+}
+
+/*
+8. Palindrome Check
+   - Define a function `isPalindrome(str)` that checks if `str` is the same
+     forwards and backwards (normalize the string and compare it to reversed version of itself).
+   - If it is, log: "<str> is a palindrome"
+   - Otherwise, log: "<str> is not a palindrome"
+*/
+
+function isPalindrome(str) {
+  let normalized = str.toLowerCase().replace(/[^a-z0-9]/g, '');
+  let reversed = normalized.split('').reverse().join('');
+  if (normalized == reversed) {
+    console.log(str + " is a palindrome");
+  }
+  else {
+    console.log(str + " is not a palindrome");
+  }
+}
+
+/*
+10. Check Even or Odd (if-else)
+   - Define a function `evenOrOdd(number)` that:
+     - Logs "Even" if the number is even
+     - Logs "Odd" if the number is odd
+*/
+
+function evenOrOdd(number) {
+  if (number % 2 == 0) {
+    console.log("Even");
+  } else {
+    console.log("Odd");
+  }
+}
+
+/*
+11. URL Protocol Checker
+   - Define a function `checkProtocol(url)` that converts the URL to lowercase
+     and checks if it starts with "https" using .startsWith().
+   - Log "Secure connection" if true, otherwise "Unsecure connection".
+*/
+
+function checkProtocol(url) {
+  if (url.toLowerCase().startsWith("https"))
+    console.log("Secure connection");
+  else
+    console.log("Unsecure connection");
+}
+/*
+13. Repeat a String
+   - Define a function `repeatWord(word, times)` that uses the .repeat() method
+     to repeat `word` `times` times.
+   - Log the repeated result.
+*/
+
+function repeatWorld(world.times) {
```

Comment:
2 errors below: 1. spelling error, 2. parameters of the function should be coma separated. You can catch errors like this easily by just running the function.
```suggestion
function repeatWord(word, times) {
```